### PR TITLE
Fixes double-counting in overlapFeatures sum

### PR DESCRIPTION
**Pull request by avmey**
`overlapFeatures()` would double count a feature if two sketches overlapped with it.

Now overlapFeatures maintains a set of features that have been overlapped when calculating the sum of individual sketches:
https://github.com/seasketch/geoprocessing/blob/9900f465a8d5e411c13ac28f026bb18583c8110e/packages/geoprocessing/src/toolbox/overlapFeatures.ts#L59

This set is then used to calculate the collection level sum:
https://github.com/seasketch/geoprocessing/blob/9900f465a8d5e411c13ac28f026bb18583c8110e/packages/geoprocessing/src/toolbox/overlapFeatures.ts#L99-L107

`overlapFeatures.test.ts` now includes a simple sum test.

### CODE REVIEW

File: packages/geoprocessing/src/toolbox/overlapFeatures.ts [hunk 1]
```diff
@@ -163,43 +190,41 @@ const getSketchPolygonIntersectArea = (
...
-      } else {
-        count = 1;
-      }
-      return {
-        count,
-      };
+
+      if (!rem) return { count: 0 };
+
+      indices.push(index);
```
Comment (twelch) [inline packages/geoprocessing/src/toolbox/overlapFeatures.ts:221]: I haven't reviewed this thoroughly but at first glance it looks like the index always gets pushed for every feature in featuresB.  Is there a reason then that this tracking should go down in this function vs. at a higher level in the stack?  I was imagining it going up in overlapFeatures.

File: packages/geoprocessing/src/toolbox/overlapFeatures.ts [hunk 2]
```diff
@@ -56,11 +56,33 @@ export async function overlapFeatures(
...
   let sumValue: number = 0;
   let isOverlap = false;
+  let featureIndices: Set<number> = new Set();
   const sketches = (Array.isArray(sketch) ? sketch : toSketchArray(sketch)).map(
     (s) => truncate(s)
   );
   const finalFeatures = features.map((f) => truncate(f));
 
+  // Create individual sketch metrics
+  const sketchMetrics: Metric[] = sketches.map((curSketch) => {
```
Comment (twelch) [inline packages/geoprocessing/src/toolbox/overlapFeatures.ts:66]: For example, could you add the second `index` parameter that map provides here and use it to track the indices?  Same as below, just a higher level

File: packages/geoprocessing/src/toolbox/overlapFeatures.ts [hunk 3]
```diff
@@ -163,43 +190,41 @@ const getSketchPolygonIntersectArea = (
...
-      return {
-        count,
-      };
+
+      if (!rem) return { count: 0 };
+
+      indices.push(index);
+
+      if (!sumProperty) return { count: 1 };
+      else if (curFeature.properties![sumProperty] >= 0)
```
Comment (twelch) [inline packages/geoprocessing/src/toolbox/overlapFeatures.ts:224]: It seems these first two conditions could be condensed into one condition
`if (sumProperty && curFeature.properties![sumProperty] >= 0)`

The !sumProperty case would just fall through to the return of count 1

File: packages/geoprocessing/src/toolbox/overlapFeatures.ts [hunk 4]
```diff
@@ -56,11 +56,33 @@ export async function overlapFeatures(
...
   let isOverlap = false;
+  let featureIndices: Set<number> = new Set();
   const sketches = (Array.isArray(sketch) ? sketch : toSketchArray(sketch)).map(
     (s) => truncate(s)
   );
   const finalFeatures = features.map((f) => truncate(f));
 
+  // Create individual sketch metrics
+  const sketchMetrics: Metric[] = sketches.map((curSketch) => {
+    const sketchValue = doIntersect(
```
Comment (twelch) [inline packages/geoprocessing/src/toolbox/overlapFeatures.ts:67]: The return is now more complex, and the variable name `sketchValue` seems incorrect?  What about `intersections`?

**Review by twelch — CHANGES_REQUESTED** [review]
Looking pretty good, some rename recommendations

File: packages/geoprocessing/src/toolbox/overlapFeatures.ts (hunk 4 above)
Comment (avmey) [reply packages/geoprocessing/src/toolbox/overlapFeatures.ts:67]: I like `intersections` as the new variable name.

**Review by twelch — APPROVED** [review]
Approving in anticipation of requested changes

**Pull request merged**